Code/Engine/Foundation/Algorithm/Implementation/HashingUtils_inl.h

```diff
@@ -120,7 +120,14 @@ struct ezHashHelper<T*>
 {
   EZ_ALWAYS_INLINE static ezUInt32 Hash(T* value)
   {
-    return ezHashHelper<size_t>::Hash(reinterpret_cast<size_t>(value) >> 4);
+    if constexpr (sizeof(T*) == 8)
```

Comment:
size_t does not map to any ez typedef on linux (it's unsigned long it or something).

Reply:
Makes totally sense, size_t is not well defined enough. Though I don't know the guarantees of "if constexpr", is it optimized in debug builds as well? I would go for #if EZ_ENABLED(EZ_PLATFORM_64BIT) to be sure.